### PR TITLE
clear cldnn unit test issue

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/gather.cpp [hunk 1]
```diff
@@ -30,8 +30,9 @@ layout gather_inst::calc_output_layout(gather_node const& node) {
         output_type = node.get_fused_output_layout().data_type;
     }
 
-    if (desc->batch_dim == 0) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/gather.cpp:33]: I think that you might try to use Gather(data, indices, axis, batch_dims) with dummy data, indices, axis nodes. 
However lets go with current fix.

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 2]
```diff
@@ -338,9 +338,6 @@ bool program_node::is_dynamic() const {
             return true;
     }
 
-    if (is_type<reshape>() && get_dependencies().size() == 2)
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:341]: could you explain about this change

**Review by yeonbok — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/reshape.cpp [hunk 3]
```diff
@@ -149,13 +149,18 @@ void reshape_inst::update_shape() {
     auto& node = const_cast<reshape_node&>(dynamic_cast<const reshape_node&>(_node));
 
     if (_node.get_dependencies().size() == 2) {
-        auto shape_mem = _network.get_output_memory(_node.get_dependency(1).id());
+        auto in_node = _node.get_dependency(1).id();
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/src/graph/reshape.cpp:152]: nit: how about changing to `in_node_id` for code readability?

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/reshape.cpp:152]: I will change it in next commit. thanks.

**Pull request merged**